ipatests/test_integration/test_nfs.py

```diff
@@ -0,0 +1,252 @@
+
+        users = {
+            "athena": "p",
+            "euripides": "s"
+        }
+        temp_pass = 'temppass'
+        for user in users:
+            self.master.run_command([
+                "ipa", "user-add",
+                "%s" % user, "--first", "%s" % user,
+                "--last", "%s" % users[user],
+                '--password'], stdin_text=temp_pass
+            )
+            self.master.run_command(["kdestroy", "-A"])
+            password = "Secret123"
+            user_kinit = "%s\n%s\n%s\n" % (temp_pass, password, password)
+            self.master.run_command(
+                ['kinit', user], stdin_text=user_kinit
+            )
+            self.master.run_command(["kdestroy", "-A"])
+            tasks.kinit_admin(self.master)
+
+    def test_krb5_nfsd(self):
+
+        nfssrv = self.clients[0]
+
+        # NFS keytab management
+        self.master.run_command([
+            "ipa", "service-add", "nfs/%s" % nfssrv.hostname
+        ])
+        nfssrv.run_command([
+            "ipa-getkeytab", "-p", "nfs/%s" % nfssrv.hostname,
+            "-k", "/etc/krb5.keytab"
+        ])
+        nfssrv.run_command(["systemctl", "restart", "nfs-server"])
+        nfssrv.run_command(["systemctl", "enable", "nfs-server"])
+        time.sleep(WAIT_AFTER_INSTALL)
+
+        # NFS share
+        # TODO use a better way to associate server-side exports
+        #      with client-side mountpoints and options
+        basedir = "exports"
+        exports = {
+            "krbnfs": "*(sec=krb5p,rw)",
+            "stdnfs": "*(ro)",
+            "home": "*(sec=krb5p,rw)"
+        }
+        for export in exports:
+            exportpath = os.sep.join(('', basedir, export))
+            exportfile = os.sep.join((
+                '', 'etc', 'exports.d', "%s.exports" % export
+            ))
+            exportline = " ".join((exportpath, exports[export]))
+            nfssrv.run_command(["mkdir", "-p", exportpath])
+            nfssrv.run_command(["chmod", "770", exportpath])
+            nfssrv.put_file_contents(exportfile, exportline)
+            nfssrv.run_command(["cat", exportfile])
+        nfssrv.run_command(["exportfs", "-r"])
+        nfssrv.run_command(["exportfs", "-s"])
+
+    def test_krb5_nfs_manual_configuration(self):
+
+        nfssrv = self.clients[0]
+        nfsclt = self.replicas[0]
+
+        nfsclt.run_command(["systemctl", "restart", "rpc-gssd"])
+        time.sleep(WAIT_AFTER_INSTALL)
+        mountpoints = ("/mnt/krb", "/mnt/std", "/home")
+        for mountpoint in mountpoints:
+            nfsclt.run_command(["mkdir", "-p", mountpoint])
+        nfsclt.run_command([
+            "systemctl", "status", "gssproxy"
+        ])
+        nfsclt.run_command([
+            "systemctl", "status", "rpc-gssd"
+        ])
+        # nfsclt.run_command([
+        #    "mount", "-t", "nfs4", "-o", "ro",
+        #    "%s:/exports/stdnfs" % nfssrv.hostname, "/mnt/std", "-v"
+        # ])
```

Comment:
This is commented code. If it is very useful for debugging purposes, please, write one comment above it, some thing like: "For debugging." If not, just remove it, please.

Reply:
Removed.